### DISCUSSION TITLE
Nginx Unit is the best to make a media server?

### DISCUSSION THREAD

**Discussion by dnegrisolli** [nginx/unit#1244] [General]
Hello!
I'm starting a new project and we need to make a home server like Azure Blobs to videos and images. Because of high costing specially in traffic, we decided to move forward to make our own media server.
Is Nginx Unit the best way to implemente something like this?
Or should I follow the classic nginx server, and add tokens to it?
Thank you

**Comment by lcrilly** [comment]
Hi @dnegrisolli,

If you could be more specific about your goals we can better advise on Unit vs nginx for this use case. What exactly do you want the server to do?

Things like auth, caching and UUID-to-filename mapping will have an impact.

**Reply by dnegrisolli** [reply]
Helo @lcrilly ! 
Thank you for your fast comment.
My goal is to make something similar to Azure Blob Storage for images and videos. The idea is to have a token(based in time) to get the url from the media, so the backend service could get the image/video.

Yes, I'll need a way to autorize the services to get the URLs to avoid that someone with a URL can access the media outside the application. 
I need to think about a automated backup to the files too.

All our infrastructure is made with Kubernetes.

**Reply by dnegrisolli** [reply]
UUID-to-filename - our backend already does it, this logic don´t need to be in the media server.

**Reply by lcrilly** [reply]
So you're looking to serve static assets if the client provides a valid token that grants access to each file?

The authentication and authorization requirements would be better-suited to nginx+[auth_request](https://nginx.org/en/docs/http/ngx_http_auth_request_module.html). Unit currently does not provide auth controls.

**Reply by dnegrisolli** [reply]
Yes, a static server with auth request. Thank you, I´ll take a look on this!
